File: 8_kyu/Find_Nearest_square_number.lua
return {
  nearest_sq = function(n)
      if not (n > 0) then
        -- error exit
        return 0
      end
      root = math.sqrt(n)
      upper = math.ceil(root)^2
      lower = math.floor(root)^2
      if upper-n < n-lower then
        return upper
      else
        return lower
      end
    end
  }

File: 8_kyu/Find_Nearest_square_number.cs
using System;

public static class Kata
{
  // Returns the nearest square number.
  // Means a number that is a square of an integer and has minimal difference to the parameter n.
  public static int NearestSq(int n) => (int) Math.Pow(Math.Round(Math.Sqrt(n)),2);
}
